enhancements/microshift/kubernetes-for-device-edge.md

```diff
@@ -0,0 +1,515 @@
+orchestration runtime is binary compatible with OpenShift Container
+Platform, but it is not 100% API resource compatible.  It has a strict
+subset of OpenShift API resources pertinent for IoT and Edge computing
+scenarios with a strong bias to retaining traditional runtime security
+controls like SecurityContextConstraints and SELinux.
+
+## Motivation
+
+MicroShift addresses customer use cases with low-resource,
+field-deployed edge devices (SBCs, SoCs) requiring a minimal K8s
+container orchestration layer.
+
+MicroShift is targeting a class of devices like Xilinx ZYNQ
+UltraScale+, fitlet2, NVIDIA Jetson TX2, or Intel NUCs that are
+pervasive in edge deployments. These cost a few hundred USD, have all
+necessary accelerators and I/O on-board, but are typically not
+extensible and are highly constrained on every resource, e.g.:
+
+* CPU: ARM Cortex-A35-class or Intel Atom class CPU, 2-4 cores, 1.5GHz clock
+* memory: 2-16GB RAM
+* storage: e.g. SATA3 @ 6Gb/s, 10kIOPS, 1ms (vs. NVMe @ 32Gb/s,
+  500kIOPS, 10µs)
+* network: Less bandwidth and less reliable than data center-based
+  servers, including being completely disconnected for extended
+  periods. Likely 1Gb/s NICs, potentially even LTE/4G connections at
+  5-10Mbps, instead of 10/25/40Gb/s NICs
+
+These use cases are not being addressed by RHEL or OpenShift
+today. RHEL doesn’t include Kubernetes. OpenShift is fundamentally
+designed for a cloud-centric and server-centric world, with
+operational models and assumptions about environmental constraints
+that are very different from field-deployed devices. Even single-node
+OpenShift deployments target traditional multi-socket data center
+computing platforms, not the small edge devices targeted by
+MicroShift.
+
+### User Stories
+
+An application developer builds applications.
+
+An application deployer packages and distributes those applications.
+
+A device administrator manages the lifecycle of a device running the
+applications.
+
+1. As an application developer, I can design my application deployment
+   so that it works with MicroShift and all other topologies of
+   OpenShift in a consistent way.
+2. As an application deployer, I can package MicroShift, the
+   applications, and any other platform components into a "system image" that
+   is written onto many devices by the manufacturer.
+3. As an application deployer, I can package MicroShift, the
+   applications, and any other platform components into a "system image" that
+   I install myself.
+4. As a device administrator, I can release an updated system image over the
+   network to upgrade devices already in the field using minimal
+   network bandwidth.
+5. As a device administrator, I can specify application-specific logic
+   with which the device will verify a successful upgrade and roll
+   back the upgrade automatically if verification fails.
+6. As a device administrator, I have control over the operating
+   system running under MicroShift.
```

Comment:
I am not sure I understand the context behind this use case. Can you give more context to this use case?

Reply:
It's up to the device administrator to configure things like add-on devices, kernel drivers, agents, etc. MicroShift does not own the OS and does not provide tools for managing it via kubernetes APIs in the way that OCP does.

Reply:
I added some more text to make this goal clearer.